src/phoenix/experimental/evals/models/litellm.py

```diff
@@ -1,125 +1,127 @@
+        try:
+            import litellm
+            from litellm import validate_environment
+
+            self._litellm = litellm
+            env_info = validate_environment(self._litellm.utils.get_llm_provider(self.model_name))
+
+            if not env_info["keys_in_environment"]:
+                raise RuntimeError(
+                    f"Missing environment variable(s): '{str(env_info['missing_keys'])}', for "
+                    f"model: {self.model_name}. \nFor additional information about the right "
+                    "environment variables for specific model providers:\n"
+                    "https://docs.litellm.ai/docs/completion/input#provider-specific-params."
+                )
+        except ImportError:
+            self._raise_import_error(
+                package_display_name="LiteLLM",
+                package_name="litellm",
+            )
+
+    def _init_model_encoding(self) -> None:
+        from litellm import decode, encode
+
+        if self.model_name in self._litellm.model_list:
+            self._encoding = encode
+            self._decoding = decode
+        else:
+            raise ValueError(
+                f"Model name not found in the LiteLLM's models list: \n{self._litellm.model_list}"
+            )
+
+    @property
+    def max_context_size(self) -> int:
+        context_size = self.max_content_size or self._litellm.get_max_tokens(self.model_name).get(
+            "max_tokens", None
+        )
+
+        if context_size is None:
+            raise ValueError(
+                "Can't determine maximum context size. An unknown model name was "
+                + f"used: {self.model_name}."
+            )
+
+        return context_size
+
+    @property
+    def encoder(self) -> "Encoding":
+        raise NotImplementedError
+
+    def get_tokens_from_text(self, text: str) -> List[int]:
+        result: List[int] = self._encoding(model=self.model_name, text=text)
+        return result
+
+    def get_text_from_tokens(self, tokens: List[int]) -> str:
+        return str(self._decoding(model=self.model_name, tokens=tokens))
+
+    async def _async_generate(self, prompt: str, **kwargs: Dict[str, Any]) -> str:
+        return self._generate(prompt, **kwargs)
+
+    def _generate(self, prompt: str, **kwargs: Dict[str, Any]) -> str:
+        messages = self._get_messages_from_prompt(prompt)
+        return str(
+            self._generate_with_retry(
+                model=self.model_name,
+                messages=messages,
+                temperature=self.temperature,
+                max_tokens=self.max_tokens,
+                top_p=self.top_p,
+                num_retries=self.num_retries,
+                request_timeout=self.request_timeout,
+                **self.model_kwargs,
+            )
+        )
+
+    def _generate_with_retry(self, **kwargs: Any) -> Any:
+        # Bypassing LiteLLM retry logic in preference of our own
+        kwargs.pop("num_retries", None)
```

Comment:
The change in line feeds has obscured the diff. This is the only line I've added: it removes the automatic LiteLLM retries in preference of our own